docs/pages/hdmi_cec_source_tests.md

```diff
@@ -0,0 +1,179 @@
+- `L2` - Level 2 Testing ()
+- `L3` - Level 3 Testing ()
+- `Sink device` - Any piece of equipment or technology that provides an input signal or data to another device or system.
+- `Source Device` - Any piece of equipment or technology that receives an input signal or data from another device or system.
+
+## Scope
+
+This document defines the requirements for testing the HDMI CEC device from a level 2/3 based on being a source device.
+
+## Overview
+
+Consumer Electronics Control (CEC) is a single-wire bidirectional bus within an HDMI system, facilitating communication among interconnected products. HDMI-CEC establishes a protocol enabling high-level control functions between audiovisual devices linked via an HDMI network, facilitating communication and control among them. Communication can occur in either Direct messaging mode or Broadcast mode. 
+
+## HDMI-CEC RDK HAL Functionality
+
+The HAL layers within RDK serve as a bridge between the underlying low-level SoC drivers and the higher-level RDK layers that utilize the functionality offered by these HAL functions. Specifically concerning the CEC Module, the HAL layers facilitate the following functionalities:
+
+- Logical Address discovery 
+- Get Logical address
+- Get Physical address
+- Syncronous transmission, and communicating via hotplug connectivity
+
+## Test scenarios
+
+The HAL CEC layer facilitates the transmission and reception of CEC information on the CEC bus. It does not handle any specific opcode commands, nor does it validate supported HAL CEC opcodes for sending or receiving.
+
+It is the responsibility of the caller to manage the opcodes. The current test cases will verify responses from connected devices for a subset of opcodes as part of the testing process.
+
+|S.No.|Test Functionality|Description|
+|-----|------------------|-----------|
+| 1 |[Logical address](#logical-address-discovery)|Facilitating the Discovery of logical addresses getting logical address of the device (for source devices) |
+| 2| [Physical Address](#physical-address)| Retrieving the physical address |
+| 3| [CEC Synchronous Transmission](#cec-synchronous-transmission)| Transmitting CEC frames and reporting on their acknowledgement |
+| 4| [CEC Receive functionality](#cec-receive-functionality)| Receiving CEC information from other devices and passing it to the layer above through a registered callback function |
+| 5| [CEC HotPlug Functionality](#cec-hotplug-functionality)| Managing CEC during Hotplug and HotUnplug events |
+
+-----------
+
+## Logical Address Discovery
+
+|S.No.|Test Functionality|Description|HAL APIs|L2|L3|Control plane requirements|
+|-----|------------------|-----------|--------|--|--|--------------------------|
+| 1 |[Logical address](#logical-address-discovery)|Get the logical address discovered during cec open and validate the address for a proper playback/tuner device. This will add the logical address, as per source functionality. |HdmiCecGetLogicalAddress|Y|NA
```

Comment:
Should provide the valid values that our platforms will get so that the test can be validated.  This is important because the test should know what values needs to be validated.

Also, the test should describe when more than one device of similar type is connected in the network and read the address.  This should be part of L3.  Also test should get no logical address found if connected more than 4 or 5 devices of same type, again as part of L3.

Reply:
Updated as requested.